### PR TITLE
docs(en): add cycle sort

### PR DESCRIPTION
**Pull request by mohitchakraverty**
**PR Checklist:**
- [x] My submission is formatted according to the guidelines in the [contributing guide](https://github.com/MakeContributions/DSA#contribution-guidelines)
- [x] My addition is on refer on the language README.md file
- [x] My addition does not have a spelling problem
- [x] My submission has a proper and user-friendly description of the algorithm
- [x] My submission has the time complexity of the algorithm
- [x] My submission has sample input-output of the program (NOT FOR PYTHON)

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
I have added Cycle Sort Algorithm description in the docs folder

**Other information:**

### CODE REVIEW

**Comment by mohitchakraverty** [comment]
@ming-tsai Waiting for your approval  :-)

File: docs/en/Sorting/Cycle-Sort.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+Again rotate rest cycle that start with index '0' and item = 2
+
+arr[] = {2, 3,  5, 10}  
+
+Above is one iteration for cycle_stat = 0.
+
+Repeat above steps for cycle_start = 1, 2, ..n-2
+
+## Implmentation
```
Comment (ming-tsai) [inline docs/en/Sorting/Cycle-Sort.md:79]: Fix typo
```suggestion
## Implementation
```

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Hi @mohitchakraverty,
I have some suggestions for this pull request:
1. Add the file index for `./docs/en/README.md`
2. Fix the typo

File: docs/en/Sorting/Cycle-Sort.md (hunk 1 above)
Comment (mohitchakraverty) [reply docs/en/Sorting/Cycle-Sort.md]: Done. Waiting for your approval :)

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**